### PR TITLE
fix: Batch markAsRead & markAsDelivered requests

### PR DESCRIPTION
**Pull request by sravan-s**
~Adding tests, can review code in the meantime~

markAsRead and markAsDelivered are called sequentially from SDK
This causes issue because, when low paying user fetch 20 channel or something
this throws network errors

* Ideally, this should be handled in SDK side, batching the requests

Fixes: https://sendbird.atlassian.net/browse/UIKIT-3791, https://sendbird.atlassian.net/browse/UIKIT-3993

### CODE REVIEW

File: src/lib/hooks/schedulerFactory.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      queue.push(channel);
+    } else {
+      logger.info('Channel: Mark as read already in queue', { channel });
+    }
+    // start the interval if it's not already running
+    if (interval) {
+      return;
+    }
+    const item = queue.shift();
+    cb(item as T);
```
Comment (sravan-s) [inline src/lib/hooks/schedulerFactory.ts:29]: if there is no item on the queue, execuete immediately

File: src/lib/hooks/schedulerFactory.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    interval = setInterval(() => {
+      if (queue.length === 0 && interval) {
```
Comment (sravan-s) [inline src/lib/hooks/schedulerFactory.ts:31]: if queue is empty, clear timeout

File: src/lib/hooks/useMarkAsDeliveredScheduler.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  }), []);
+
+  useEffect(() => {
+    // for simplicity, we clear the queue when the connection is lost
+    if (!isConnected) {
+      markAsDeliveredScheduler.clear();
+    }
+  }, [isConnected]);
+
+  useUnmount(() => { markAsDeliveredScheduler.clear(); });
```
Comment (sravan-s) [inline src/lib/hooks/useMarkAsDeliveredScheduler.ts:45]: to clear the queue when SendbirdProvider is unmounted

File: src/modules/App/stories/index.stories.js [hunk 4]
```diff
@@ -53,8 +54,8 @@ export const versionInfo = () => {
 export const basicSDK = () => fitPageSize(
   <App
     appId={appId}
-    userId={userId}
-    nickname={userId}
+    userId={uuidv4()}
+    nickname={uuidv4()}
```
Comment (sravan-s) [inline src/modules/App/stories/index.stories.js:58]: Will remove before merge ~

File: src/modules/ChannelList/components/ChannelListUI/index.tsx [hunk 5]
```diff
@@ -127,12 +128,12 @@ const ChannelListUI: React.FC<ChannelListUIProps> = (props: ChannelListUIProps)
...
                 ?.find((feature) => (feature === DELIVERY_RECIPT));
 
-              if (canSetMarkAsDelivered) {
+              if (canSetMarkAsDelivered && !disableMarkAsDelivered) {
                 logger.info('ChannelList: Marking all channels as read');
                 // eslint-disable-next-line no-unused-expressions
                 channelList?.forEach((channel) => {
                   if (channel?.unreadMessageCount > 0) {
-                    markAsReadScheduler.push(channel);
+                    markAsDeliveredScheduler.push(channel);
```
Comment (sravan-s) [inline src/modules/ChannelList/components/ChannelListUI/index.tsx:136]: This was a bug, we should have called `markAsDelivered` when fetching channel list

File: src/lib/hooks/schedulerFactory.ts (hunk 1 above)
Comment (HoonBaek) [reply src/lib/hooks/schedulerFactory.ts:35]: here's no condition to check if there's no item

**Review by HoonBaek — APPROVED** [review]

**Pull request merged**